### DISCUSSION TITLE
Procedure for releasing on PyPI

### DISCUSSION THREAD

**Discussion by diazona** [pytest-dev/pytest-localserver#16] [General]
I think all our planned changes for v0.5.1 are in the repo, so I wanted to make sure we're in sync about the procedure for releasing the new version to PyPI. If I haven't missed anything, I think these should be the steps:

1. Merge a commit that bumps the version numbers in `setup.py` and `pytest_localserver/__init__.py`
2. After that's merged, tag the head of `master` as `v0.5.1`
3. Maybe GPG-sign the tag? :shrug: I'm not sure if we care about this
4. Build sdist and wheel packages from the tagged commit using e.g. `pyproject-build -sw` or `python setup.py sdist bdist_wheel`
5. Upload the packages to PyPI

Anything else?

FWIW I don't particularly care who does it - I don't mind following these steps, but if one of you @redtoad or @coordt want to do it, go right ahead, once we're all agreed on the procedure.

**Comment by redtoad** [comment]
Go for it.

**Reply by diazona** [reply]
Cool. I'll give it a little while just in case @coordt wants to chime in, and then I'll plan to make the release on the 18th.

**Reply by diazona** [reply]
I'm making the release now

**Comment by diazona** [comment]
Version bump PR is #17

**Comment by coordt** [comment]
I'm fine with the procedure. I do have some ideas for automating it in the future. I'll pull some notes together and put them in an issue or discussion.

**Reply by diazona** [reply]
Yeah that'd be great! In other (non-public) projects I've worked on I have the whole procedure automated so I fully appreciate the value of doing so :smile: but all my experience is with GitLab, not Github.

**Comment by diazona** [comment]
Just uploaded and https://pypi.org/project/pytest-localserver/ reflects the new version :tada:
